content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,598 @@
+download_data('./data/', "ru")
+download_data('./data/', "mx")
+
+us_data = pickle.load(open("./data/us.p", "rb"))  # 100,000 examples
+gb_data = pickle.load(open("./data/gb.p", "rb"))  # 100,000 examples
+ru_data = pickle.load(open("./data/ru.p", "rb"))  # 100,000 examples
+mx_data = pickle.load(open("./data/mx.p", "rb"))  # 100,000 examples
+
+dataset_vectorizer.vectorize(us_data)
+dataset_vectorizer.vectorize(gb_data)
+dataset_vectorizer.vectorize(ru_data)
+dataset_vectorizer.vectorize(mx_data)
+```
+
+##### First Test
+
+Now let's test for the United States of America and United Kingdom.
+
+```python
+us_loader = DataLoader(us_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(us_loader)
+exp_bi_lstm.test(us_loader)
+
+gb_loader = DataLoader(gb_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(gb_loader)
+exp_bi_lstm.test(gb_loader)
+```
+
+The next table presents the results of both models for both countries. We obtain
+better results for the two countries using the bidirectional bi-LSTM (around 8% better). It's interesting to see that, considering address structures are similar to those in the training dataset (Canada), we obtain near as good results as those observed during training. This suggests that our model seems to have learned to recognize the structure of an address. Also, despite the language being the same as in the training dataset (i.e. some English address in the bilingual canadian address dataset), we obtain poorer results. That situation is most likely due to the fact that the postal code formats are not the same. For the US, it is 5 digits, and for the UK it is similar to that of Canada, but it is not always a letter followed by a number and not always 6 characters. It is *normal* for a model to
+have difficulty when faced with new patterns. All in all, we can say that our model has achieved good results.
+
+| Model (Country) | LSTM one layer | Bidirectional bi-LSTM |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (US)    |     0.6176     |       **0.3078**   |
+|  Accuracy (US)  |     84.7396    |       **91.8220**  |
+|    Loss (UK)    |     0.4368     |       **0.1571**   |
+|  Accuracy (UK)  |     86.2543    |       **95.6840**  |
+
+
+##### The Second and Third Test
+
+Now let's test for Russia and Mexico. 
+
+But first, let's discuss how our French embeddings can generate word vectors for vocabulary in a different language. FastText use subword embeddings when complete embeddings do not exist. For example, we can assume the presence of a word embedding vector for the word `Roi`, but we face an out-of-vocabulary (OOV) for the word `H1A1` since this word is not a real word. The trick with fastText is that they create compose embeddings using the subword with fixed window size (length of the subword) when facing OOV word. For example, a two characters window embeddings of `H1A1` would be the aggregated embeddings of the subword `H1`, `1A` and `A1`. 
```

Comment:
```suggestion
But first, let's discuss how our French embeddings can generate word vectors for vocabulary in a different language. FastText uses subword embeddings when complete embeddings do not exist. For example, we can assume the presence of a word embedding vector for the word `Roi`, but we face an out-of-vocabulary (OOV) for the word `H1A1` since this word is not a real word. The trick with fastText is that it creates composite embeddings using the subword with fixed window size (length of the subword) when facing OOV words. For example, a two characters window embeddings of `H1A1` would be the aggregated embeddings of the subword `H1`, `1A` and `A1`. 
```